rust/arrow/src/array/array.rs

```diff
@@ -2099,6 +2338,69 @@ mod tests {
         assert_eq!(2, list_array.value_length(1));
     }
 
+    #[test]
+    fn test_dictionary_array() {
+        // Construct a value array
```

Comment:
Isn't is unusual to create a `DictionaryArray` where the `values` array is larger than the `keys`?

Reply:
Yes, this is true. The most commonly observed dictionary arrays in our work have been two values "YES" and "NO" - don't ask why! Arrow does not support Boolean dictionaries, but that is probably not a bad thing.